File: _CodeTopics/LeetCode_contest/weekly/weekly2020/180/180_1.py
class Solution(object):
    def luckyNumbers (self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[int]
        """
        
        # 矩阵的行数
        m = len(matrix)
        # 矩阵的列数
        n = len(matrix[0])
        lucknumberlist = []
        
        #for k in range(n):
        #    mat[k].ap
        
        for i in range(m):
            curr_row = matrix[i] 
            curr_row_min = min(curr_row)
            ind = curr_row.index(curr_row_min)
            for j in range(m):
                flag = 1
                if curr_row_min < matrix[j][ind]:
                    flag = 0
                    break
            if flag == 1:
                lucknumberlist.append(curr_row_min)
        return lucknumberlist

File: _CodeTopics/LeetCode/1201-1400/001380--weekly180_1/tran/180_1.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {

        // 第 180 场周赛第一题
        // https://leetcode-cn.com/submissions/detail/53863331/
        
        int m = matrix.size(), n = matrix[0].size();
        vector<int> lucknumberlist;
        for (int i = 0; i < m; ++i) {
            vector<int> curr_row = matrix[i];
            int curr_row_min = *min_element(curr_row.begin(), curr_row.end());
            vector<int>::iterator it = find(curr_row.begin(), curr_row.end(), curr_row_min);
            int ind = std::distance(curr_row.begin(), it);
            
            bool flag = true;
            for (int j = 0; j < m; ++j) {
                if (curr_row_min < matrix[j][ind]) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                lucknumberlist.push_back(curr_row_min);
            } else {
                flag = true;
            }
        }
        return lucknumberlist;
    }
};

/*
https://leetcode-cn.com/submissions/detail/268441735/

执行用时：16 ms, 在所有 C++ 提交中击败了92.96%的用户
内存消耗：11.7 MB, 在所有 C++ 提交中击败了8.45%的用户
通过测试用例：
104 / 104
*/
